viaf.py

```diff
@@ -2,60 +2,271 @@
 from nose.tools import set_trace
 from lxml import etree
 import logging
+from fuzzywuzzy import fuzz
 import re
 
 from collections import Counter, defaultdict
 
+from core.metadata_layer import (
+    ContributorData,
+    Metadata, 
+)
+
 from core.model import (
     Contributor,
     DataSource,
     Representation,
 )
 
+from core.util.personal_names import (
+    display_name_to_sort_name, 
+    is_corporate_name, 
+    normalize_contributor_name_for_matching, 
+)
+
+from core.util.titles import (
+    normalize_title_for_matching, 
+    unfluff_title, 
+)
+
 from core.util.xmlparser import (
     XMLParser,
 )
-from core.util.personal_names import display_name_to_sort_name
+
+
+
 
 class VIAFParser(XMLParser):
 
     NAMESPACES = {'ns2' : "http://viaf.org/viaf/terms#"}
 
     log = logging.getLogger("VIAF Parser")
+    log.setLevel(logging.INFO)  # TODO: remove
+    wikidata_id = re.compile("^Q[0-9]")
+
+
+
+    @classmethod
+    def combine_nameparts(self, given, family, extra):
+        """Turn a (given name, family name, extra) 3-tuple into a
+        display name.
+        """
+        if not given and not family:
+            return None
+        if family and not given:
+            display_name = family
+        elif given and not family:
+            display_name = given
+        else:
+            display_name = given + ' ' + family
+        if extra and not extra.startswith('pseud'):
+            if family and given:
+                display_name += ', ' + extra
+            else:
+                display_name += ' ' + extra
+        return display_name
+
+
+    @classmethod
+    def contributor_name_match_ratio(cls, name1, name2):
+        """
+        Returns a number between 0 and 100, representing the percent 
+        match (Levenshtein Distance) between name1 and name2, 
+        after each has been normalized.
+        """
+        name1 = cls.prepare_contributor_name_for_matching(name1)
+        name2 = cls.prepare_contributor_name_for_matching(name2)
+        match_ratio = fuzz.ratio(name1, name2)
+        return match_ratio
+
 
     @classmethod
     def name_matches(cls, n1, n2):
+        """ Returns true if n1 and n2 are identical strings.  
+        Ignores case, periods, commas, spaces after single-letter initials.
+        """ 
```

Comment:
It doesn't really do half of these things; change the docstring and add the code later?